league_rpc_linux/processes/lcu_thread.py

```diff
@@ -0,0 +1,355 @@
+    if event.data["availability"] == "away":
+        data["availability"] = "Away"
+
+    delayUpdate()
+
+
+@connector.ws.register("/lol-gameflow/v1/gameflow-phase", event_types=("UPDATE",))
+async def gameflow_phase_updated(connection, event):
+    print("Gameflow Phase has been updated.")
+    data["gameflowPhase"] = event.data  # returns plain string of the phase
+
+    delayUpdate()
+
+
+@connector.ws.register("/lol-champ-select/v1/session", event_types=("DELETE",))
+async def champ_select_closed(connection, event):
+    data["inChampSelect"] = False
+    delayUpdate()
+
+
+@connector.ws.register("/lol-champ-select/v1/session", event_types=("CREATE",))
+async def champ_select_joined(connection, event):
+    data["inChampSelect"] = True
+    delayUpdate()
+
+
+# could be used for lobby instead: /lol-gameflow/v1/gameflow-metadata/player-status
+@connector.ws.register(
+    "/lol-lobby/v2/lobby", event_types=("UPDATE", "CREATE", "DELETE")
+)
+async def in_lobby(connection, event):
+    print(f"Lobby Data - {event.type}")
+
+    if event.type == "Delete":
+        data["inLobby"] = False
+        delayUpdate()
+        return
+
+    data["inLobby"] = True
+
+    if event.type == "Create":
+        # wait for Update Call, as create does not contain data
+        return
+
+    data["queue_id"] = event.data["gameConfig"]["queueId"]
+    data["lobby_id"] = event.data["partyId"]
+    data["players"] = len(event.data["members"])
+    data["max_players"] = int(event.data["gameConfig"]["maxLobbySize"])
+    data["map_id"] = event.data["gameConfig"]["mapId"]
+    data["gamemode"] = event.data["gameConfig"]["gameMode"]
+    data["isCustom"] = event.data["gameConfig"]["isCustom"]
+    if event.data["gameConfig"]["gameMode"] == "PRACTICETOOL":
+        data["isPractice"] = True
+        data["max_players"] = 1
+
+    if data["queue_id"] == -1:
+        # custom game / practice tool / tutorial lobby
+        if data["isPractice"]:
+            data["queue"] = "Practice Tool"
+        else:
+            data["queue"] = "Custom Game"
+        delayUpdate()
+        return
+
+    lobbyQueueInfoRaw = await connection.request(
+        "GET", "/lol-game-queues/v1/queues/" + str(data["queue_id"])
+    )
+    lobbyQueueInfo = await lobbyQueueInfoRaw.json()
+
+    data["queue"] = lobbyQueueInfo["name"]
+    data["lobbyQueueIsRanked"] = lobbyQueueInfo["isRanked"]
+
+    delayUpdate()
+
+
+# ranked stats
+@connector.ws.register("/lol-ranked/v1/current-ranked-stats/", event_types=("UPDATE",))
+async def ranked(connection, event):
+    data["summoner_rank"] = (
+        event.data["highestRankedEntry"]["tier"]
+        + " "
+        + event.data["highestRankedEntry"]["division"]
+        + ": "
+        + str(event.data["highestRankedEntry"]["leaguePoints"])
+        + " LP"
+    )
+
+    delayUpdate()
+
+
+# ranked stats
+
+
+## queue
+@connector.ws.register("/lol-matchmaking/v1/search", event_types=("CREATE",))
+async def matchmaking(connection, event):
```

Comment:
Since this function name is defined twice, could we change it to better fit the purpose of the function?

Reply:
consolidated into a single funtion